internal/internal_worker.go

```diff
@@ -230,10 +230,10 @@ func ensureRequiredParams(params *workerExecutionParameters) {
 // verifyNamespaceExist does a DescribeNamespace operation on the specified namespace with backoff/retry
 // It returns an error, if the server returns an EntityNotExist or BadRequest error
 // On any other transient error, this method will just return success
-func verifyNamespaceExist(client workflowservice.WorkflowServiceClient, namespace string, logger log.Logger) error {
+func verifyNamespaceExist(client workflowservice.WorkflowServiceClient, metricsScope tally.Scope, namespace string, logger log.Logger) error {
 	ctx := context.Background()
 	descNamespaceOp := func() error {
-		grpcCtx, cancel := newGRPCContext(ctx)
+		grpcCtx, cancel := newGRPCContext(ctx, grpcMetricsScope(metrics.GetEmptyRPCScope(metricsScope)))
```

Comment:
Instead of passing `metrics.GetEmptyRPCScope(metricsScope)` every time you can just set it in `intenal_utils.go` line 130 as:
```
MetricsScope: metrics.GetEmptyRPCScope(metricsScope),
```
and it will be overridden every time you pass specific metrics scope.

Reply:
The problem is GRPCContext is not aware of rootMetricsScope.

Reply:
You're right. The only thing I don't like here is that the option which is passed every time is not an option but required parameter. Let's keep it as is and I might refactor it later.